circle.yml

```diff
@@ -1,6 +1,19 @@
 machine:
   ruby:
     version: 2.2.2
+  services:
+    - docker
 test:
   override:
     - bundle exec rake --trace
+deployment:
+  create_containers:
+    branch: master
+    commands:
+      # Log into the docker container. Note that this requires these environment variables
+      # to be stored in the CircleCI environment config
+      - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
+      # Ensure that the docker container built contains absolutely nothing unexpected
+      # that might have been installed by CircleCI (eg an updated database.yml)
+      - (git clean -fd ; git reset --hard) || true
```

Comment:
It's probably best for us to actually generate an error if git clean or reset can't do their job. Let me fix that. Thanks.